src/qonnx/custom_op/ChannelsLast/wrapped_ops.py

```diff
@@ -0,0 +1,488 @@
+import numpy as np
+import onnxruntime as rt
+from copy import deepcopy
+from google.protobuf.pyext._message import RepeatedScalarContainer
+from onnx import TensorProto, helper
+
+from finn.custom_op.base import CustomOp
+from finn.custom_op.general.im2col import compute_conv_output_dim
+from finn.custom_op.general.maxpoolnhwc import compute_pool_output_dim
+
+
+class ChannelsLastWrappedOp(CustomOp):
+    # ToDo: _channelsLast_node_types should be loaded / inferred from this file or the registry.
+    # Standard ONNX nodes which require a ChannelsLast data format to function properly
+    _channelsLast_node_types = ["Conv", "MaxPool", "BatchNormalization"]
+    # Required for ChannelsLast transformations and ops
+    # Transpose parameters to convert to channels last for 3D and 4D tensors
+    _to_chan_last_args = {
+        3: (0, 2, 1),
+        4: (0, 2, 3, 1),
+    }
+    # Similarly for converting back to channels first.
+    _to_chan_first_args = {
+        3: (0, 2, 1),
+        4: (0, 3, 1, 2),
+    }
```

Comment:
consider making a utility function to return the axis permutation order (both ways, chanfirst -> chanlast and chanlast -> chanfirst) given the number of dimensions

chanfirst -> chanlast : create list(range(ndim)) then move the dimension with ind "1" to the final position
chanlast -> chanfirst : create list(range(ndim)) then move the dimension with ind "ndim-1" to the next-first position

Reply:
Okay, I moved this into functions in the `base_wrapped_op.py` file.